### DISCUSSION TITLE
Optimizing the performance of a web driver

### DISCUSSION THREAD

**Discussion by admiral504** [omkarcloud/google-maps-scraper#41] [Q&A]
I have a question about optimizing the performance of a web driver, and I believe you might be able to help.

I've heard that adding certain options to the code can improve the performance of the web driver. However, I'm not entirely sure how to go about adding these options. Here's the list of options I've come across:

--no-sandbox
--disable-dev-shm-usage
--disable-renderer-backgrounding
--disable-background-timer-throttling
--disable-backgrounding-occluded-windows
--disable-client-side-phishing-detection
--disable-crash-reporter
--disable-oopr-debug-crash-dump
--no-crash-upload
--disable-gpu
--disable-extensions
--disable-low-res-tiling
--log-level=3
--silent

Could you kindly guide me on how to incorporate these options into the code? Your expertise in this matter would be greatly appreciated.

**Comment by Chetan11-dev** [comment]
"Premature optimization is the root of all evil."
                                    - Donald Knuth
                                    
So, something that works, is better than perfect.

**Additionally,** one of the most critical optimizations is already in place – disabling images, fonts, and CSS:

![Screenshot (4)](https://github.com/omkarcloud/google-maps-scraper/assets/53407137/e6f27154-2407-4406-911a-8f50d7fa7b5d)

Moreover, many of the optimizations you're suggesting are already incorporated into the Bose Framework.

![Screenshot (2)](https://github.com/omkarcloud/google-maps-scraper/assets/53407137/b98bf87e-5b02-4013-ac75-a15fc10ddf06)

Additionally, if you still wish to add options (though it's unlikely to make any noticeable difference), you can override the create_driver method in the BaseTask class and return a driver with the desired options.

Peace :)
